src/common-ssh/ssh.c

```diff
@@ -445,9 +539,20 @@ guac_common_ssh_session* guac_common_ssh_create_session(guac_client* client,
      * https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp2906.pdf
      */
     if (guac_fips_enabled()) {
-        libssh2_session_method_pref(session, LIBSSH2_METHOD_KEX, FIPS_COMPLIANT_KEX_ALGORITHMS);
-        libssh2_session_method_pref(session, LIBSSH2_METHOD_CRYPT_CS, FIPS_COMPLIANT_CIPHERS);
-        libssh2_session_method_pref(session, LIBSSH2_METHOD_CRYPT_SC, FIPS_COMPLIANT_CIPHERS);
+        char* kex_str = get_method_pref_string(client,
+            session, LIBSSH2_METHOD_KEX, FIPS_COMPLIANT_KEX_ALGORITHMS);
+        libssh2_session_method_pref(session, LIBSSH2_METHOD_KEX, kex_str);
+        guac_mem_free(kex_str);
+
+        char* crypt_cs_str = get_method_pref_string(client,
+            session, LIBSSH2_METHOD_CRYPT_CS, FIPS_COMPLIANT_CIPHERS);
+        libssh2_session_method_pref(session, LIBSSH2_METHOD_CRYPT_CS, crypt_cs_str);
+        guac_mem_free(crypt_cs_str);
+      
+        char* crypt_sc_str = get_method_pref_string(client,
+            session, LIBSSH2_METHOD_CRYPT_SC, FIPS_COMPLIANT_CIPHERS);
+        libssh2_session_method_pref(session, LIBSSH2_METHOD_CRYPT_SC, crypt_sc_str);
+        guac_mem_free(crypt_sc_str);
```

Comment:
According to https://libssh2.org/libssh2_session_method_pref.html, the `libssh2_session_method_pref()` function will ignore any preferred cipers/algorithms/etc. if not supported by libssh2:

> ... If a method is listed which is not supported by libssh2 it will be ignored and not sent to the remote host during protocol negotiation.

So, in case this helps simplify things, it's not necessary to dynamically assemble a string based on what libssh2 has available (though it's still helpful to log whether something isn't supported).